web/src/settings_org.ts

```diff
@@ -231,30 +227,38 @@ function set_msg_edit_limit_dropdown(): void {
 }
 
 function message_move_limit_setting_enabled(
-    related_setting_name: "realm_edit_topic_policy" | "realm_move_messages_between_streams_policy",
+    related_setting_name:
+        | "realm_edit_topic_policy"
+        | "realm_can_move_messages_between_channels_group",
 ): boolean {
-    const setting_value_string = $<HTMLSelectOneElement>(
-        `select:not(multiple)#id_${CSS.escape(related_setting_name)}`,
-    ).val();
-    assert(setting_value_string !== undefined);
-    const setting_value = Number.parseInt(setting_value_string, 10);
-
-    let settings_options;
     if (related_setting_name === "realm_edit_topic_policy") {
-        settings_options = settings_config.edit_topic_policy_values;
-    } else {
-        settings_options = settings_config.move_messages_between_streams_policy_values;
-    }
-
-    if (setting_value === settings_options.by_admins_only.code) {
-        return false;
-    }
+        const setting_value_string = $<HTMLSelectOneElement>(
+            `select:not(multiple)#id_${CSS.escape(related_setting_name)}`,
+        ).val();
+        assert(setting_value_string !== undefined);
+        const setting_value = Number.parseInt(setting_value_string, 10);
+        const settings_options = settings_config.edit_topic_policy_values;
+
+        if (
+            setting_value === settings_options.by_admins_only.code ||
+            setting_value === settings_options.by_moderators_only.code ||
+            setting_value === settings_options.nobody.code
+        ) {
+            return false;
+        }
 
-    if (setting_value === settings_options.by_moderators_only.code) {
-        return false;
+        return true;
     }
-
-    if (setting_value === settings_options.nobody.code) {
+    const user_group_id = settings_components.get_dropdown_list_widget_setting_value(
+        $(`#id_${related_setting_name}`),
+    );
+    assert(typeof user_group_id === "number");
+    const user_group_name = user_groups.get_user_group_from_id(user_group_id).name;
+    if (
+        user_group_name === "role:administrators" ||
+        user_group_name === "role:moderators" ||
+        user_group_name === "role:nobody"
+    ) {
         return false;
     }
```

Comment:
@sahil839 can you review this PR, but especially this section?

Reply:
This might change soon though when we allow anonymous groups, like we would need to check whether all the allowed users are admins or moderators. Or we can check it like it only has the admins, moderators system groups and nothing else.